<!doctype html>
<html>
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <title>eigenMT by joed3</title>

    <link rel="stylesheet" href="stylesheets/styles.css">
    <link rel="stylesheet" href="stylesheets/github-light.css">
    <meta name="viewport" content="width=device-width">
    <!--[if lt IE 9]>
    <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
  </head>
  <body>
    <div class="wrapper">
      <header>
        <h1>eigenMT</h1>
        <p>An Efficient Multiple-Testing Adjustment for eQTL Studies that Accounts for Linkage Disequilibrium between Variants</p>

        <p class="view"><a href="https://github.com/joed3/eigenMT">View the Project on GitHub <small>joed3/eigenMT</small></a></p>


        <ul>
          <li><a href="https://github.com/joed3/eigenMT/zipball/master">Download <strong>ZIP File</strong></a></li>
          <li><a href="https://github.com/joed3/eigenMT/tarball/master">Download <strong>TAR Ball</strong></a></li>
          <li><a href="https://github.com/joed3/eigenMT">View On <strong>GitHub</strong></a></li>
        </ul>
      </header>
      <section>
        <h1>
<a id="eigenmt" class="anchor" href="#eigenmt" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>eigenMT</h1>

<p>An Efficient Multiple-Testing Adjustment for eQTL Studies that Accounts for Linkage Disequilibrium between Variants</p>

<h2>
<a id="introduction" class="anchor" href="#introduction" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Introduction</h2>

<p>eigenMT is a computationally efficient multiple testing correction method for cis-eQTL studies. Typically, modern cis-eQTL studies correct for multiple testing across thousands of variants for a single gene using permutations. This correction method, though accurate, requires a high computational cost on the order of days to weeks for large numbers of permutations and/or large sample sizes. Our method reduces this computational burden by orders of magnitude while maintaining high accuracy when compared to permutations. To accomplish this, our method attempts to estimate the number of effective tests performed for a given gene by directly accounting for the LD relationship among the tested variants.  </p>

<h2>
<a id="download" class="anchor" href="#download" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Download</h2>

<p>Begin by downloading or cloning this respository. eigenMT runs as a stand-alone python script. It requires an active installaton of python (version 2.7 or higher) to be installed. To download and install a python distribution, there are a few convenient options:</p>

<ul>
<li><a href="https://store.continuum.io/cshop/anaconda/">Anaconda</a></li>
<li><a href="https://www.enthought.com/products/canopy/">Enthought Canopy</a></li>
</ul>

<p>These bundled installations already include a number of modules required for running eigenMT including:</p>

<ul>
<li><a href="http://www.numpy.org/">numpy</a></li>
<li><a href="http://www.scipy.org/">scipy</a></li>
<li><a href="http://scikit-learn.org/stable/">Scikit-learn</a></li>
</ul>

<p>Again, these modules should come pre-packaged in one of the bundled python installations above. If you decide to install these packages yourself, please see the listed websites for detailed instructions.</p>

<h2>
<a id="input" class="anchor" href="#input" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Input</h2>

<p>A typical run will have the following command line format:</p>

<pre><code>python eigenMT.py --CHROM &lt;chromosome ID&gt;
    --QTL &lt;Matrix-eQTL SNP-gene tests file&gt; \
    --GEN &lt;Matrix-eQTL genotype matrix&gt; \
    --POS &lt;Matrix-eQTL genotype position file&gt; \
    --var_thresh [variance explained threshold, default 0.99] \
    --OUT &lt;output filename&gt; \
    --window [window size, default 200]
</code></pre>

<p>The input fields are defined as follows.</p>

<table>
<thead>
<tr>
<th>Argument</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>CHROM</td>
<td>Chromosome ID. Indicates which chromosome the analysis will be performed on. Must match the ID used in the Matrix-eQTL SNP-gene tests file.</td>
</tr>
<tr>
<td>QTL</td>
<td>Filename for Matrix-eQTL SNP-gene tests file. See <code>example/qtls.txt</code> for an example.</td>
</tr>
<tr>
<td>GEN</td>
<td>Genotype matrix in Matrix-eQTL format. Can accept either hardcoded or dosage based genotypes. Missing genotypes must be encoded as NA and will be imputed to the mean genotype during correction. See <code>example/genotypes.txt</code> for an example.</td>
</tr>
<tr>
<td>GENPOS</td>
<td>Matrix-eQTL genotype position file. See <code>example/positions.txt</code> for an example.</td>
</tr>
<tr>
<td>var_thresh</td>
<td>Threshold for amount of variance explained in the genotype correlation matrix. Default is 99% variance explained. Increasing this threshold will increase estimates of effective number of tests (M_eff) and decrease accuracy of the approximation.</td>
</tr>
<tr>
<td>OUT</td>
<td>Output filename. Output format is described below.</td>
</tr>
<tr>
<td>window</td>
<td>Window size parameter. Determines what size of disjoint windows to split genotype matrices for each gene into. Default is 200 SNPs. We recommend using a window size of at least 50 SNPs up to 200 SNPs to balance accuracy and speed.</td>
</tr>
<tr>
<td>cis_dist</td>
<td>Threshold for bp distance from the gene TSS to perform multiple testing correction. Default is 1e6.</td>
</tr>
<tr>
<td>external</td>
<td>Logical indicating whether the provided genotype matrix is different from the one originally used to test for cis-eQTLs.</td>
</tr>
</tbody>
</table>

<h2>
<a id="output" class="anchor" href="#output" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Output</h2>

<p>The output file is in tab-separated format with the following columns.</p>

<table>
<thead>
<tr>
<th>Column</th>
<th>Description</th>
</tr>
</thead>
<tbody>
<tr>
<td>1</td>
<td>variant ID</td>
</tr>
<tr>
<td>2</td>
<td>Gene ID</td>
</tr>
<tr>
<td>3</td>
<td>T-statistic from Matrix-eQTL</td>
</tr>
<tr>
<td>4</td>
<td>p-value from Matrix-eQTL</td>
</tr>
<tr>
<td>5</td>
<td>FDR estimate from Matrix-eQTL</td>
</tr>
<tr>
<td>6</td>
<td>estimate of effect size BETA from Matrix-eQTL</td>
</tr>
<tr>
<td>7</td>
<td>eigenMT corrected p-value</td>
</tr>
<tr>
<td>8</td>
<td>estimated number of independent tests for the gene</td>
</tr>
</tbody>
</table>

<p>Note: The first 5 columns of the output will correspond to the typical Matrix-eQTL output. Also, each tested gene will appear once in the output file with its most significant SNP and the eigenMT corrected p-value.</p>

<h2>
<a id="example" class="anchor" href="#example" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Example</h2>

<p>We offer a small example dataset for testing. We provide a genotype matrix and corresponding position file in Matrix-eQTL format. This genotype matrix is for chromosome 19 for the EUR373 samples as part of the <a href="http://www.nature.com/nature/journal/v501/n7468/full/nature12531.html?WT.ec_id=NATURE-20130926">GEUVADIS</a> study. We also provide a sample of the cis-eQTL tests performed for these samples using Matrix-eQTL. This sample SNP-gene tests file represents 100 genes. To extract the example dataset, run</p>

<pre><code>tar -zxvf example.tar.gz
</code></pre>

<p>To run eigenMT on the example data, use the following command:</p>

<pre><code>python eigenMT.py --CHROM 19 \
    --QTL example/qtls.txt \
    --GEN example/genotypes.txt \
    --GENPOS example/gen.positions.txt \
    --PHEPOS example/phe.positions.txt \
    --OUT example/exampleOut.txt
</code></pre>

<p>Note: this example uses the default settings for window size and variance explained threshold.</p>

<h2>
<a id="external-genotype-data" class="anchor" href="#external-genotype-data" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>External genotype data</h2>

<p>Our method offers the ability to perform multiple testing correction using a genotype matrix from a separate sample population than the one initially used for cis-eQTL testing. It is important to note that this external genotype matrix should come from the same background population as the one under study. For example, if cis-eQTL testing is performed in samples of European ancestry, then any external genotype matrix used should come from a similar European population. We have shown that using genotype data from studies with larger sample sizes can improve the accuracy of our method compared to using the genotype data for the study. Genotype data from larger studies will provide better estimates of the LD structure for variants around each gene, improving our estimates of the effective number of tests.</p>

<h2>
<a id="population-stratification-and-other-covariates" class="anchor" href="#population-stratification-and-other-covariates" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Population stratification and other covariates</h2>

<p>We recommend first removing the effects of population stratification (genotype PCs, population or ancestry assignments) and other covariates (age, gender, PEER factors, etc) from the expression matrix. We have shown that by first removing these effects and then performing cis-eQTL calling on the inverse rank normalized residuals, we maintain conservativeness and accuracy. We will soon provide an R script with example code for how to perform this normalization step prior to a run of eigenMT. </p>

<h2>
<a id="citation" class="anchor" href="#citation" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Citation</h2>

<p>Davis JR, Fresard L, Knowles DA, Pala M, Bustamante CD, Battle A, Montgomery SB (2015). An Efficient Multiple-Testing Adjustment for eQTL Studies that Accounts for Linkage Disequilibrium between Variants. The American Journal of Human Genetics, 98(1), 216–224. doi: (<a href="http://doi.org/10.1016/j.ajhg.2015.11.021">http://doi.org/10.1016/j.ajhg.2015.11.021</a>)</p>

<h2>
<a id="contact" class="anchor" href="#contact" aria-hidden="true"><span aria-hidden="true" class="octicon octicon-link"></span></a>Contact</h2>

<ul>
<li>Joe Davis: <a href="mailto:joed3@stanford.edu">joed3@stanford.edu</a>
</li>
<li>Laure Fresard: <a href="mailto:lfresard@stanford.edu">lfresard@stanford.edu</a>
</li>
<li>Stephen Montgomery: <a href="mailto:smontgom@stanford.edu">smontgom@stanford.edu</a>
</li>
</ul>
      </section>
      <footer>
        <p>This project is maintained by <a href="https://github.com/joed3">joed3</a></p>
        <p><small>Hosted on GitHub Pages &mdash; Theme by <a href="https://github.com/orderedlist">orderedlist</a></small></p>
      </footer>
    </div>
    <script src="javascripts/scale.fix.js"></script>
    
  </body>
</html>
